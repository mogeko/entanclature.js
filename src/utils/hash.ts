import crypto from "crypto";

function hash(msg: Buffer) {
  const sha256 = crypto.createHash("sha256").update(msg).digest("hex");

  return sha256.substring(0, 7).toUpperCase();
}

export default hash;

if (import.meta.vitest) {
  const { it, expect } = import.meta.vitest;
  const exampleText = "example";
  const exampleHash = "50D858E";

  it("hash", () => {
    const _hash = hash(Buffer.from(exampleText, "utf-8"));

    expect(_hash).toEqual(exampleHash);
    expect(_hash).toMatchSnapshot();
  });
}
