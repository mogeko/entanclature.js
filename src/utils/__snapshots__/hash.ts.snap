// Vitest Snapshot v1

exports[`hash 1`] = `"50D858E"`;
